server/src/main/resources/db/changelog/20191218180815-create-product-and-provided-product-mapping.xml

```diff
@@ -0,0 +1,63 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<databaseChangeLog
+        xmlns="http://www.liquibase.org/xml/ns/dbchangelog"
+        xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+        xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog
+        http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-3.1.xsd">
+
+    <changeSet id="20191218180815-1" author="sdhome">
+        <comment>Create table for product and provided/derived-provided products mapping.</comment>
+
+        <createTable tableName="cp2_product_provided_products">
+            <column name="product_uuid" type="VARCHAR(255)">
+                <constraints nullable="false"/>
+            </column>
+
+            <column name="provided_prod_uuid" type="VARCHAR(255)">
+                <constraints nullable="false"/>
+            </column>
+
+            <column name="derived_provided_prod_uuid" type="VARCHAR(255)">
```

Comment:
We can't put both of these in the same table unless we have a discriminator column, especially if both are marked as not accepting null.

These will need to be modeled either after the current provided and derived provided product tables associated with pools, or the single table we used in the past which used a discriminator value.

Reply:
I misunderstood that Products have Provided Products. And Provided Products have derived Provided products.  After a discussion with @abhiskum, I understood that products and derived products are different entities. They have their respective provided products.

I'll keep the separate two tables for individual mappings.